### ISSUE TITLE
Infinite recursion is happening while merging two dicts in CM

### ISSUE THREAD

**Issue by arjunsuresh** [mlcommons/ck#858]
```
cm  run script --tags=generate-run-cmds,inference,_find-performance,_all-scenarios --model=gptj-99 --implementation=reference --device=cuda --backend=pytorch --category=edge --division=open --quiet --precision=bfloat16 -v
```

The above command is hanging

**Comment by arjunsuresh** [comment]
This is a [potential fix](https://github.com/ctuning/mlcommons-ck/commit/ab900f2294de7de425808e91ad69d2b9adb86547) for this.

**Comment by gfursin** [comment]
It should be fixed in the v1.5.1 release. Thank you!
